common/base/include/claragenomics/utils/device_preallocated_allocator.cuh

```diff
@@ -221,8 +227,11 @@ private:
                                                    });
         assert(block_to_be_freed_iter != std::end(used_blocks_));
 
-        // ** wait for all work on associated_stream to finish before freeing up this memory block
-        CGA_CU_ABORT_ON_ERR(cudaStreamSynchronize(block_to_be_freed_iter->associated_stream));
+        // ** wait for all work on associated_streams to finish before freeing up this memory block
+        for (const cudaStream_t& associated_stream : block_to_be_freed_iter->associated_streams)
```

Comment:
Optional: Just use `cudaStream_t`, since it is only a tiny handle. Furthermore, doing a Synchronize on a `const` argument is a bit confusing imo.

Reply:
Makes sense, I think we can be rather sure `cudaSream_t` won't explode in the future.
done